packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -1069,12 +1158,56 @@ class _DialState extends State<_Dial> with SingleTickerProviderStateMixin {
           accentColor: themeData.accentColor,
           theta: _theta.value,
           activeRing: _activeRing,
-        )
+          textDirection: Directionality.of(context),
+        ),
       )
     );
   }
 }
 
+class _DialPaint extends CustomPaint {
+  const _DialPaint({
+    @required this.dialPainter,
+  }) : super(
+         key: const ValueKey<String>('time-picker-dial'), // used for testing.
+         painter: dialPainter,
+       );
+
+  final _DialPainter dialPainter;
+
+  @override
+  RenderCustomPaint createRenderObject(BuildContext context) {
+    return new _RenderDialPaint(
+      initialDialPainter: dialPainter,
+    );
+  }
+}
+
+class _RenderDialPaint extends RenderCustomPaint {
+  _RenderDialPaint({
+    _DialPainter initialDialPainter,
+  }) : super(painter: initialDialPainter);
+
+  @override
+  void describeSemanticsConfiguration(SemanticsConfiguration config) {
+    super.describeSemanticsConfiguration(config);
+    config.isSemanticBoundary = true;
+    config.textDirection = TextDirection.ltr;
```

Comment:
why ltr? aren't the arabic labels rtl?

Reply:
Good catch. This is left-over from my experiments. Removed.